core/src/main/scala/org/scalatra/servlet/FileUploadSupport.scala

```diff
@@ -73,6 +74,7 @@ trait FileUploadSupport extends ServletBase with HasMultipartConfig {
    * `IllegalStateException` for some other reason.
    */
   protected def isSizeConstraintException(e: Exception): Boolean = e match {
+    case exc: ServletException => exc.getRootCause.getMessage == "400: bad multipart"
```

Comment:
Is this related to upgrading to Jetty12? Otherwise, better to separate PR.

Reply:
It is indeed related to the Jetty 12.
See https://github.com/jetty/jetty.project/blob/jetty-12.0.x/jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletApiRequest.java#L591 for the reason of this change.

Reply:
Got it. Thanks for clarification.

Reply:
I have added some comments regarding specificities related to Jetty 10 and 12.
(I've also refined the way how those exceptions are "recognized".)

Reply:
Thank you!